packages/beacon-node/src/chain/archiver/interface.ts

```diff
@@ -0,0 +1,46 @@
+import {CheckpointWithHex} from "@lodestar/fork-choice";
+import {Metrics} from "../../metrics/metrics.js";
+import {RootHex} from "@lodestar/types";
+
+export enum ArchiveMode {
+  Full = "full",
```

Comment:
More of a general question, once we have differential archive mode implemented, will there be any advantage to use our current (full) archiving strategy?

If not, we should probably not make it overly generic / switchable

Reply:
My idea is to get the differential strategy released as experimental first. As I am a bit skeptical about it's memory implication. So during that time user can choose which strategy to use.